src/atoms/SegmentControl/__tests__/SegmentControl.native.test.js

```diff
@@ -0,0 +1,231 @@
+          <SegmentControl.Option value="Option 3" testID="segment-option-3" />
+        </SegmentControl>,
+      );
+      const option = getByTestId('segment-option-1');
+
+      act(() => {
+        fireEvent.pressIn(option);
+      });
+      expect(container).toMatchSnapshot();
+    });
+
+    it('filled segment control unselected', () => {
+      const { container, getByTestId } = renderWithTheme(
+        <SegmentControl defaultValue="Option 1" variant="filled" size="small">
+          <SegmentControl.Option value="Option 1" testID="segment-option-1" />
+          <SegmentControl.Option value="Option 2" testID="segment-option-2" />
+          <SegmentControl.Option value="Option 3" testID="segment-option-3" />
+        </SegmentControl>,
+      );
+      const option = getByTestId('segment-option-2');
+
+      act(() => {
+        fireEvent.pressIn(option);
+      });
+      expect(container).toMatchSnapshot();
+    });
+    it('filled segment control selected', () => {
+      const { container, getByTestId } = renderWithTheme(
+        <SegmentControl defaultValue="Option 1" variant="filled" size="small">
+          <SegmentControl.Option value="Option 1" testID="segment-option-1" />
+          <SegmentControl.Option value="Option 2" testID="segment-option-2" />
+          <SegmentControl.Option value="Option 3" testID="segment-option-3" />
+        </SegmentControl>,
+      );
+      const option = getByTestId('segment-option-1');
+
+      act(() => {
+        fireEvent.pressIn(option);
+      });
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  describe('onPress', () => {
+    it('outlined segment control ', () => {
+      const { container, getByTestId } = renderWithTheme(
+        <SegmentControl defaultValue="Option 1" variant="outlined" size="small">
+          <SegmentControl.Option value="Option 1" testID="segment-option-1" />
+          <SegmentControl.Option value="Option 2" testID="segment-option-2" />
+          <SegmentControl.Option value="Option 3" testID="segment-option-3" />
+        </SegmentControl>,
+      );
+      const option = getByTestId('segment-option-2');
+
+      act(() => {
+        fireEvent.press(option);
+      });
+      expect(container).toMatchSnapshot();
+    });
+    it('filled segment control ', () => {
+      const { container, getByTestId } = renderWithTheme(
+        <SegmentControl defaultValue="Option 1" variant="filled" size="small">
+          <SegmentControl.Option value="Option 1" testID="segment-option-1" />
+          <SegmentControl.Option value="Option 2" testID="segment-option-2" />
+          <SegmentControl.Option value="Option 3" testID="segment-option-3" />
+        </SegmentControl>,
+      );
+      const option = getByTestId('segment-option-2');
+
+      act(() => {
+        fireEvent.press(option);
+      });
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  describe('disabled', () => {
```

Comment:
Looks like you have missed test cases for `subText` prop, may be for which an expectation in the test case would have caught the typo `0` in the code. :P
F man, I have started talking TDD language! 😆

Reply:
Hahahaha! Thanks 🙏